pkg/app/types.go

```diff
@@ -109,10 +112,21 @@ type EncryptionService interface {
 	Decrypt(ciphertext []byte) ([]byte, error)
 }
 
+type Session struct {
+	Key         string
+	OAuth2State string
+}
+
 type DatabaseService interface {
 	// Credentials are usually stored encrypted hence the []byte type.
 	// If no credentials are available for the given user Fetch returns nil, nil.
 	FetchBuildAPICredentials(username string) ([]byte, error)
 	// Store new credentials or overwrite existing ones for the given user.
 	StoreBuildAPICredentials(username string, credentials []byte) error
+	// Create or update a user session.
+	CreateOrUpdateSession(s *Session) error
```

Comment:
ideally this object would be immutable, as it's not the case wondering whether we could receive a copy instead so we don't risk the object getting modified before being stored.

Reply:
Done